iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -997,6 +1012,66 @@ public void postGetTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
     }
   }
 
+  @Override
+  public void preDropPartitions(org.apache.hadoop.hive.metastore.api.Table hmsTable,
+      EnvironmentContext context,
+      List<org.apache.commons.lang3.tuple.Pair<Integer, byte[]>> partExprs)
+      throws MetaException {
+    Table icebergTbl = IcebergTableUtil.getTable(conf, hmsTable);
+    DeleteFiles deleteFiles = icebergTbl.newDelete();
+    List<Expression> expressions = partExprs.stream().map(partExpr -> {
+      ExprNodeDesc exprNodeDesc = SerializationUtilities
+          .deserializeObjectWithTypeInformation(partExpr.getRight(), true);
+      SearchArgument sarg = ConvertAstToSearchArg.create(conf, (ExprNodeGenericFuncDesc) exprNodeDesc);
+      return HiveIcebergFilterFactory.generateFilterExpression(sarg);
+    }).collect(Collectors.toList());
+    PartitionsTable partitionsTable = (PartitionsTable) MetadataTableUtils
+        .createMetadataTableInstance(icebergTbl, MetadataTableType.PARTITIONS);
+    List<PartitionData> partitionList = Lists.newArrayList();
+    Expression finalExp = Expressions.alwaysFalse();
+    PartitionSpec pSpec = icebergTbl.spec();
+    for (int index = 0; index < expressions.size(); index++) {
+      finalExp = Expressions.or(finalExp, expressions.get(index));
+    }
+    try (CloseableIterable<FileScanTask> fileScanTasks = partitionsTable.newScan().planFiles()) {
+      fileScanTasks.forEach(task ->
+          partitionList.addAll(Sets.newHashSet(CloseableIterable.transform(task.asDataTask().rows(), row -> {
+            StructProjection data = row.get(0, StructProjection.class);
+            PartitionData partitionData = IcebergTableUtil.toPartitionData(data, pSpec.partitionType());
+            boolean required = false;
+            for (Expression expr : expressions) {
+              ResidualEvaluator resEval = ResidualEvaluator.of(icebergTbl.spec(), expr, false);
+              if (resEval.residualFor(partitionData).isEquivalentTo(Expressions.alwaysTrue())) {
+                required = true;
```

Comment:
just return partitionData here, no need for `required `

Reply:
We need to filter the partitions which do not satisfy the conditions.

Reply:
Done